### PR TITLE
emcee 3.1.4 [fix py38] :snowflake: 

### PR DESCRIPTION
**Pull request by lorepirri**
emcee 3.1.4 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-5381]
- dev_url (pypi): https://github.com/dfm/emcee/tree/v3.1.4
- conda-forge: https://github.com/conda-forge/emcee-feedstock/blob/main/recipe/meta.yaml
- pypi: https://pypi.org/project/emcee/3.1.4
- pypi inspector: https://inspector.pypi.io/project/emcee/3.1.4

### Explanation of changes:

- add py38, add upstream tests, bump build number

[PKG-5381]: https://anaconda.atlassian.net/browse/PKG-5381?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,16 +22,21 @@ requirements:
...
   imports:
     - emcee
-  commands:
-    - pip check
   requires:
     - pip
+    - pytest
+  commands:
+    - pip check
+    - pytest -v --pyargs emcee
```
Comment (ifitchet) [inline recipe/meta.yaml:36]: Does this need a `[not (...)]` selector?

**Review by ifitchet — APPROVED** [review]
LGTM

File: recipe/meta.yaml (hunk 1 above)
Comment (lorepirri) [reply recipe/meta.yaml:36]: True, I forgot to remove a leftover. Thanks!

File: recipe/meta.yaml [hunk 2]
```diff
@@ -22,16 +22,18 @@ requirements:
     - setuptools-scm
     - wheel
   run:
-    - numpy
     - python
+    - {{ pin_compatible('numpy') }}
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:26]:
```suggestion
    - numpy
```
As it doesn't rely on `numpy` in `host` then we can use a plain `numpy` as it's defined in the requirements https://github.com/dfm/emcee/blob/v3.1.4/setup.py#L25. `pin_compatible` is used when you have numpy in host. That will get correct pinnings for different Python versions at runtime

Comment (lorepirri) [reply recipe/meta.yaml:26]: I'll try without pinning

**Review by skupr-anaconda — APPROVED** [review]

**Review by psteyer — APPROVED** [review]
LGTM!

**Pull request merged**